### PR TITLE
RD-2110 RD-2112 Add kinesis support

### PR DESCRIPTION
**Pull request by doriaviram**

### CODE REVIEW

File: src/main/java/io/lumigo/core/parsers/KinesisParser.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            List<String> messageIds = extractMessageIds(response.getAwsResponse());
+            if (messageIds.size() > 0) span.getInfo().setMessageIds(messageIds);
+        } catch (Exception e) {
+            Logger.error(e, "Failed to extract parse for Kinesis request");
+        }
+    }
+
+    private List<String> extractMessageIds(Object response) {
+
+        ArrayList<String> result = new ArrayList<>();
```
Comment (uri-p) [inline src/main/java/io/lumigo/core/parsers/KinesisParser.java]:
```suggestion
        List<String> result = new LinkedList<>();
```

File: src/test/java/io/lumigo/core/parsers/KinesisParserTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void test_parse_kinesis_parser_with_no_data() {
+        when(request.getParameters()).thenReturn(new HashMap<>());
+
+        kinesisParser.parse(span, request, new Response(null, httpResponse));
+
+        assertNull(span.getInfo().getResourceName());
```
Comment (uri-p) [inline src/test/java/io/lumigo/core/parsers/KinesisParserTest.java]: Compare full object

File: src/test/java/io/lumigo/core/parsers/KinesisParserTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    @Test
+    void test_parse_kinesis_put_record_simple_flow() {
+        response = new Response(putRecordResult, httpResponse);
+        when(putRecordResult.getSequenceNumber())
+                .thenReturn("fee47356-6f6a-58c8-96dc-26d8aaa4631a");
+        when(putRecordRequest.getStreamName()).thenReturn("streamName");
+        when(request.getOriginalRequest()).thenReturn(putRecordRequest);
+        kinesisParser.parse(span, request, response);
```
Comment (uri-p) [inline src/test/java/io/lumigo/core/parsers/KinesisParserTest.java]: add \n

File: src/main/java/io/lumigo/core/parsers/KinesisParser.java (hunk 1 above)
Comment (doriaviram) [reply src/main/java/io/lumigo/core/parsers/KinesisParser.java]: 👍

File: src/test/java/io/lumigo/core/parsers/KinesisParserTest.java (hunk 2 above)
Comment (doriaviram) [reply src/test/java/io/lumigo/core/parsers/KinesisParserTest.java]: 👍

File: src/test/java/io/lumigo/core/parsers/KinesisParserTest.java (hunk 3 above)
Comment (doriaviram) [reply src/test/java/io/lumigo/core/parsers/KinesisParserTest.java]: 👍

**Pull request merged**